server/middleware/per-request.js

```diff
@@ -31,6 +31,16 @@ function perRequestContextFactory(options) {
   var scope = options.name || name;
   var enableHttpContext = options.enableHttpContext || false;
   var ns = LoopBackContext.createContext(scope);
+  // Monkey-patch LoopBackContext.getCurrentContext in order to fix issue #17
```

Comment:
Regarding the concurrent requests, even if the code needs to be fixed, tests pass with the workaround (and fail if I comment out the workaround), and it already works in a test app with 10 concurrent `curl` requests.

I have a possible explanation. Are you sure that points 1 and 3 are correct? In these points, the function that gets invoked is this:
https://github.com/strongloop/loopback-context/blob/v1.0.0/server/middleware/per-request.js#L36
*This* function does *not* modify the global LoopBackContext.getCurrentContext (it's the outer function that does, and the latter gets called when *adding* the middleware to the app).